### PR TITLE
Fix: check current step instance before calling end step

### LINKED ISSUE #6523: useCucumberStepReporter: true is giving warning of allure-js-commons: Unexpected endStep(). There are no any steps running
**Environment:**
 - **WebdriverIO version:** [7.0.7]
 - **Mode:** [WDIO Runner]
 - **If WDIO Testrunner, running sync/async:** [async]
 - **Node.js version:** [v12.18.3]
 - **NPM version:** [6.14.6]
 - **Browser name and version:** [Chrome 88.0.4324.192]
 - **Platform name and version:** [MacOS Catalina]
 - **Additional wdio packages used (if applicable):** [@wdio/allure-reporter, @wdio/spec reporter, @wdio/selenium-standalone service, @wdio/browserstack-service, @wdio/cucumber-framework, @wdio/devtools-service, @wdio/spec-reporter, @wdio/local-runner]
 
**Describe the bug**
If `useCucumberStepReporter: true` is set in the reporter, allure is giving the warning of `allure-js-commons: Unexpected endStep(). There are no any steps running`

### PR DESCRIPTION
**Pull request by iamkenos**
## Proposed changes

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

Fixes #6523 

Add instance checks on current step before calling `endStep`

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

N/A

### CODE REVIEW

File: packages/wdio-allure-reporter/src/index.ts [hunk 1]
```diff
@@ -211,7 +211,10 @@ class AllureReporter extends WDIOReporter {
     onTestPass() {
         attachConsoleLogs(this._consoleOutput, this._allure)
         if (this._options.useCucumberStepReporter) {
-            return this._allure.endStep('passed')
+            const suite = this._allure.getCurrentSuite()
+            if (suite && suite.currentStep instanceof Step) {
```
Comment (erwinheitzman) [inline packages/wdio-allure-reporter/src/index.ts:215]: Shouldn't `suite.currentStep` simply be undefined or null? This would simplify the check

**Review by erwinheitzman — COMMENTED** [review]
Thnka you for this fix, I left a small question for you :)

File: packages/wdio-allure-reporter/src/index.ts (hunk 1 above)
Comment (iamkenos) [reply packages/wdio-allure-reporter/src/index.ts:215]: Thanks for taking a look! I left some details here - https://github.com/webdriverio/webdriverio/issues/6523#issuecomment-889823216

 If currentStep is of another type, say `Test` which I believe is the cause of this issue, (not sure why that happens though, maybe hooks? maybe only on async?), It sould still satisfy a null / undef check.

Thanks!

**Comment by christian-bromann** [comment]
Thanks a lot for taking a stab at this!

Can you revert the line break changes? It makes it otherwise hard to review the test. I assume you run Windows and your IDE added a Carriage Return (CR) to the end of every line.

**Comment by iamkenos** [comment]
> Thanks a lot for taking a stab at this!
> 
> Can you revert the line break changes? It makes it otherwise hard to review the test. I assume you run Windows and your IDE added a Carriage Return (CR) to the end of every line.

Thanks for taking a look! I'm running on mac and

**Comment by staizen-alexm** [comment]
> > Thanks a lot for taking a stab at this!
> > Can you revert the line break changes? It makes it otherwise hard to review the test. I assume you run Windows and your IDE added a Carriage Return (CR) to the end of every line.
> 
> Thanks for taking a look! I'm running a mac and am using lf. I've a feeling the last fella who edited the file was on windows. Vscode git lens diff showed proper diffs so i was also surprised with the git diff. Weird. Lemme see what i can do..

@christian-bromann separated the test changes into 2 commits; you check [this](https://github.com/webdriverio/webdriverio/pull/7208/commits/74fd641d2f19ca3c03800b78f6e1be4a375f7663) for a better diff.

rebased on main too.

thanks!

**Comment by christian-bromann** [comment]
> you check [this](https://github.com/webdriverio/webdriverio/pull/7208/commits/fddb9fe4b1ffb82878a17df1882c76438e160378) for a better diff.

Thanks! Why is the main diff so large though? Can you check we didn't add any `CR` in it?

**Comment by iamkenos** [comment]
> > you check [this](https://github.com/webdriverio/webdriverio/pull/7208/commits/fddb9fe4b1ffb82878a17df1882c76438e160378) for a better diff.
> 
> Thanks! Why is the main diff so large though? Can you check we didn't add any `CR` in it?

The main diff is huge because the version of that file in `main` has cr instead of lf, must have been missed in the previous change review of that file.. I don't have access to git rn but i checked it this morning with the `file` command. The first test commit in this pr replaces tjose with lf instead.

**Review by christian-bromann — APPROVED** [review]
Indeed, this PR fixes the line break style. Thanks! Will add an ESLint rule for it.

Thanks a lot again 👍

**Pull request merged**